core/src/main/java/io/questdb/griffin/engine/functions/groupby/FirstNotNullIPv4GroupByFunctionFactory.java

```diff
@@ -64,15 +64,30 @@ public Func(@NotNull Function arg) {
         }
 
         @Override
-        public void computeNext(MapValue mapValue, Record record) {
-            if (mapValue.getIPv4(valueIndex) == Numbers.IPv4_NULL) {
-                computeFirst(mapValue, record);
+        public void computeNext(MapValue mapValue, Record record, long rowId) {
+            if (mapValue.getIPv4(valueIndex + 1) == Numbers.IPv4_NULL) {
+                computeFirst(mapValue, record, rowId);
             }
         }
 
         @Override
         public String getName() {
             return "first_not_null";
         }
+
+        @Override
+        public void merge(MapValue destValue, MapValue srcValue) {
+            int srcVal = srcValue.getIPv4(valueIndex + 1);
+            if (srcVal == Numbers.IPv4_NULL) {
+                return;
+            }
+            long srcRowId = srcValue.getLong(valueIndex);
+            long destRowId = destValue.getLong(valueIndex);
+            // srcRowId is non-null at this point since we know that the value is non-null
+            if (srcRowId < destRowId || destRowId == Numbers.LONG_NaN) {
+                destValue.putLong(valueIndex, srcRowId);
+                destValue.putInt(valueIndex + 1, srcVal);
```

Comment:
kind of assymmetic calls, getIpv4/putInt()

Reply:
That's because we don't have a put method for IPv4, but only the getter. Do you think it's ok to add it in this PR?

Reply:
I would add a method, otherwise we will never do it.